zapgrpc/zapgrpc_test.go

```diff
@@ -33,44 +34,210 @@ import (
+		logger.Printf("%s world", "hello")
+		logger.Println()
 		logger.Println("foo")
+		logger.Println("foo", "bar")
+		logger.Println("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
 	})
 }
 
 func TestLoggerDebugSuppressed(t *testing.T) {
 	checkMessages(t, zapcore.InfoLevel, []Option{WithDebug()}, zapcore.DebugLevel, nil, func(logger *Logger) {
 		logger.Print("hello")
-		logger.Printf("world")
+		logger.Printf("%s world", "hello")
+		logger.Println()
 		logger.Println("foo")
+		logger.Println("foo", "bar")
+	})
+}
+
+func TestLoggerWarningExpected(t *testing.T) {
+	checkMessages(t, zapcore.DebugLevel, nil, zapcore.WarnLevel, []string{
+		"hello",
+		"s1s21 2 3s34s56",
+		"hello world",
+		"",
+		"foo",
+		"foo bar",
+		"s1 s2 1 2 3 s3 4 s5 6",
+	}, func(logger *Logger) {
+		logger.Warning("hello")
+		logger.Warning("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
+		logger.Warningf("%s world", "hello")
+		logger.Warningln()
+		logger.Warningln("foo")
+		logger.Warningln("foo", "bar")
+		logger.Warningln("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
+	})
+}
+
+func TestLoggerErrorExpected(t *testing.T) {
+	checkMessages(t, zapcore.DebugLevel, nil, zapcore.ErrorLevel, []string{
+		"hello",
+		"s1s21 2 3s34s56",
+		"hello world",
+		"",
+		"foo",
+		"foo bar",
+		"s1 s2 1 2 3 s3 4 s5 6",
+	}, func(logger *Logger) {
+		logger.Error("hello")
+		logger.Error("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
+		logger.Errorf("%s world", "hello")
+		logger.Errorln()
+		logger.Errorln("foo")
+		logger.Errorln("foo", "bar")
+		logger.Errorln("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
 	})
 }
 
 func TestLoggerFatalExpected(t *testing.T) {
 	checkMessages(t, zapcore.DebugLevel, nil, zapcore.FatalLevel, []string{
 		"hello",
-		"world",
+		"s1s21 2 3s34s56",
+		"hello world",
+		"",
 		"foo",
+		"foo bar",
+		"s1 s2 1 2 3 s3 4 s5 6",
 	}, func(logger *Logger) {
 		logger.Fatal("hello")
-		logger.Fatalf("world")
+		logger.Fatal("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
+		logger.Fatalf("%s world", "hello")
+		logger.Fatalln()
 		logger.Fatalln("foo")
+		logger.Fatalln("foo", "bar")
+		logger.Fatalln("s1", "s2", 1, 2, 3, "s3", 4, "s5", 6)
+	})
+}
+
+func TestLoggerVTrueExpected(t *testing.T) {
+	enabled := map[zapcore.Level][]int{
+		zapcore.DebugLevel: {
+			grpcLvlInfo, grpcLvlWarn, grpcLvlError, grpcLvlFatal,
+		},
+		zapcore.InfoLevel: {
+			grpcLvlInfo, grpcLvlWarn, grpcLvlError, grpcLvlFatal,
+		},
+		zapcore.WarnLevel: {
+			grpcLvlWarn, grpcLvlError, grpcLvlFatal,
+		},
+		zapcore.ErrorLevel: {
+			grpcLvlError, grpcLvlFatal,
+		},
+		zapcore.DPanicLevel: {
+			grpcLvlFatal,
+		},
+		zapcore.PanicLevel: {
+			grpcLvlFatal,
+		},
+		zapcore.FatalLevel: {
+			grpcLvlFatal,
+		},
+	}
+	for zapLvl, grpcLvls := range enabled {
+		for _, grpcLvl := range grpcLvls {
+			t.Run(fmt.Sprintf("%s %d", zapLvl, grpcLvl), func(t *testing.T) {
+				checkLevel(t, zapLvl, true, func(logger *Logger) bool {
+					return logger.V(grpcLvl)
+				})
+			})
+		}
+	}
+}
+
+func TestLoggerVFalseExpected(t *testing.T) {
```

Comment:
suggest merging this test with the one above using a test table like:
```
tests := struct{
  zapLevel zapcore.Level
  grpcEnabled []int 
  grpcDisabled []int
}
```

Reply:
Done